src/main/java/org/opentripplanner/routing/impl/PathFinder/WalkToBikePathFinder.java

```diff
@@ -0,0 +1,162 @@
+                } else {
+                    if (graphPath.states.getFirst().getTimeSeconds() < request.dateTime) {
+                        LOG.error("A graph path leaves before the requested time. This implies a bug.");
+                        gpi.remove();
+                    }
+                }
+            }
+        }
+
+        if (paths == null || paths.size() == 0) {
+            LOG.debug("Path not found: " + request.from + " : " + request.to);
+            request.rctx.debugOutput.finishedRendering(); // make sure we still report full search time
+            throw new PathNotFoundException();
+        }
+
+        return paths;
+    }
+
+    /**
+     * Break up a RoutingRequest with intermediate bike location into two separate requests,
+     * before and after the position of the bike.
+     *
+     * If there is no bike location, issue a single request. Otherwise process the two requests
+     *  either from left to right (if {@code request.arriveBy==false})
+     * or from right to left (if {@code request.arriveBy==true}). In the latter case the order of
+     * the requested paths is (bikeLocation, to), and (from, bikeLocation).
+     */
+    DebugOutput debugOutput = null;
+
+    private List<GraphPath> getGraphPathConsideringBikeLocation (RoutingRequest request) {
+        Collection<Vertex> temporaryVertices = new ArrayList<>();
+        if (request.hasBikeLocation()) {
+            GraphPath pathFromBike;
+            GraphPath pathToBike;
+            if (request.arriveBy) {
+                pathFromBike = getGraphPathFromBike(request, temporaryVertices, request.dateTime);
+                if (pathFromBike == null) {
+                    return new ArrayList<GraphPath>();
+                }
+                pathToBike = getGraphPathToBike(request, temporaryVertices, pathFromBike.getStartTime());
+                if (pathToBike == null) {
+                    return new ArrayList<GraphPath>();
+                }
+            } else {
+                pathToBike = getGraphPathToBike(request, temporaryVertices, request.dateTime);
+                if (pathToBike == null) {
+                    return new ArrayList<GraphPath>();
+                }
+                pathFromBike = getGraphPathFromBike(request, temporaryVertices, pathToBike.getEndTime());
+                if (pathFromBike == null) {
+                    return new ArrayList<GraphPath>();
+                }
+            }
+            request.setRoutingContext(router.graph);
+            request.rctx.debugOutput = debugOutput;
+            debugOutput = null;
+            return Collections.singletonList(joinParts(pathToBike, pathFromBike));
+        } else {
+            return getPaths(request);
+        }
+    }
+
+    private GraphPath getGraphPathToBike (RoutingRequest request, Collection<Vertex> temporaryVertices, long time) {
+        RoutingRequest intermediateRequest = request.clone();
+        intermediateRequest.setNumItineraries(1);
```

Comment:
Consider merging the two functions getGraphPathToBike and FromBike, and making it a boolean parameter to decide if it's to/from. Just for better code maintanability.